Framework/Algorithms/src/CalculatePlaczekSelfScattering.cpp

```diff
@@ -95,19 +105,19 @@ void CalculatePlaczekSelfScattering::exec() {
                      neutronMass / atom.second["mass"];
   }
   // get incident spectrum and 1st derivative
-  const MantidVec xLambda = inWS->readX(0);
-  const MantidVec incident = inWS->readY(0);
-  const MantidVec incidentPrime = inWS->readY(1);
+  const MantidVec xLambda = incidentWS->readX(0);
+  const MantidVec incident = incidentWS->readY(0);
+  const MantidVec incidentPrime = incidentWS->readY(1);
   double dx = (xLambda[1] - xLambda[0]) / 2.0;
   std::vector<double> phi1;
   for (size_t i = 0; i < xLambda.size() - 1; i++) {
     phi1.push_back((xLambda[i] + dx) * incidentPrime[i] / incident[i]);
   }
   // set detector law term (eps1)
-  const double lambdaD = 1.44;
   std::vector<double> eps1;
-  for (size_t i = 0; i < xLambda.size(); i++) {
-    double xTerm = -xLambda[i] / lambdaD;
+  const double lambdaD = 1.44;
```

Comment:
The all caps rule is only for global constants, as you can see both the definition and use of the constant in the same area of the code using a different naming scheme just hurts readability.

You should, howver, be using `auto` here.

Reply:
Why use `auto` in this case instead of `double`?

Reply:
Because the type is given on the RHS of the assignment.

This case is pretty trivial, but take `const double a = b + c();`. If the type of `b` or return type of `c` were to change you may introduce unintended conversions, etc. Using `auto` will use the type that matches whatever is on the RHS.

Reply:
are you talking about the type for `xTerm` or for `LambdaD`?

Reply:
`LambdaD`, although both cases are valid.